tests/Backend/ExternalBuckets/SnowflakeExternalBucketShareTest.php

```diff
@@ -0,0 +1,183 @@
+
+    /** @var Client */
+    protected $shareClient;
+
+    /**
+     * @var Client
+     */
+    protected $linkingClient;
+
+    public function setUp(): void
+    {
+        parent::setUp();
+        $this->initEmptyTestBucketsForParallelTests();
+        $this->shareClient = $this->getClientForToken(
+            STORAGE_API_SHARE_TOKEN,
+        );
+        $this->linkingClient = $this->getClientForToken(
+            STORAGE_API_LINKING_TOKEN,
+        );
+    }
+
+    public function testShareExternalBucket(): void
+    {
+        $STAGE = 'in';
+        $BUCKET_NAME = 'test-bucket-ext-share';
+
+        $this->dropBucketIfExists($this->_client, $STAGE.'.'.$BUCKET_NAME, true);
+
+        $this->initEvents($this->_client);
+
+        $guide = $this->_client->registerBucketGuide([self::EXTERNAL_DB, self::EXTERNAL_SCHEMA], 'snowflake');
+
+        $guideExploded = explode("\n", $guide['markdown']);
+        $db = $this->ensureSnowflakeConnection();
+
+        $db->executeQuery(
+            sprintf(
+                'DROP DATABASE IF EXISTS %s;',
+                SnowflakeQuote::quoteSingleIdentifier(self::EXTERNAL_DB),
+            ),
+        );
+        $db->executeQuery(
+            sprintf(
+                'CREATE DATABASE %s;',
+                SnowflakeQuote::quoteSingleIdentifier(self::EXTERNAL_DB),
+            ),
+        );
+        $db->executeQuery(
+            sprintf(
+                'USE DATABASE %s;',
+                SnowflakeQuote::quoteSingleIdentifier(self::EXTERNAL_DB),
+            ),
+        );
+        $db->executeQuery(
+            sprintf(
+                'CREATE SCHEMA %s;',
+                SnowflakeQuote::quoteSingleIdentifier(self::EXTERNAL_SCHEMA),
+            ),
+        );
+        $db->executeQuery(
+            sprintf(
+                'USE SCHEMA %s;',
+                SnowflakeQuote::quoteSingleIdentifier(self::EXTERNAL_SCHEMA),
+            ),
+        );
+        $db->executeQuery(
+            sprintf(
+                'CREATE TABLE %s (ID INT, LASTNAME VARCHAR(255));',
+                SnowflakeQuote::quoteSingleIdentifier(self::EXTERNAL_TABLE),
+            ),
+        );
+
+        foreach ($guideExploded as $command) {
+            if (str_starts_with($command, 'GRANT') && !str_contains($command, 'FUTURE')) {
+                try {
+                    $db->executeQuery($command);
+                } catch (Exception $e) {
+                    $this->fail($e->getMessage() . ': ' . $command);
+                }
+            }
+        }
+
+        $this->_client->registerBucket(
+            $BUCKET_NAME,
+            [self::EXTERNAL_DB, self::EXTERNAL_SCHEMA],
+            $STAGE,
+            'will not fail',
+            'snowflake',
+            $BUCKET_NAME,
+        );
+
+        $tables = $this->_client->listTables($STAGE.'.'.$BUCKET_NAME);
+        $this->assertCount(1, $tables);
+        $bucket = $this->_client->refreshBucket($STAGE.'.'.$BUCKET_NAME);
```

Comment:
preco tu je ten refresh? Mozno sa to zide pak ked v testoch pribudne linkovanie, ale teraz nevidim nejaky specialny vyznam.

Reply:
refresh jen abych to prohnal connection a měl jistotu, že mám fakt správný `$bucket['id']`, co má connection

jinak bych samozřejmě mohl použít `$STAGE.'.'.$BUCKET_NAME`

Reply:
volas tam prave ` $bucket = $this->_client->refreshBucket($STAGE.'.'.$BUCKET_NAME);` alebo nechapem uplne jak to myslis :) Mozes normalne pouzit `getTable` ak chces overit ze ti to najde tu tabulku

Reply:
asi neřeš, už jsem to napsal jinak a asi lépe :-)